test/common/stats/allocator_impl_test.cc

```diff
@@ -125,6 +127,142 @@ TEST_F(AllocatorImplTest, RefCountDecAllocRaceSynchronized) {
   EXPECT_FALSE(alloc_.isMutexLockedForTest());
 }
 
+TEST_F(AllocatorImplTest, ForEachSinkedCounter) {
+
+  StatNameHashSet sinked_stat_names;
+  std::vector<CounterSharedPtr> sinked_counters;
+  std::vector<CounterSharedPtr> unsinked_counters;
+
+  alloc_.setCounterSinkFilter([&sinked_stat_names](const Stats::Counter& counter) {
+    if (sinked_stat_names.count(counter.statName()) == 1) {
+      return true;
+    }
+    return false;
+  });
+
+  size_t n_stats = 11;
```

Comment:
num_stats would be more common in our style.

Also this is const, so declare it that way.